src/main/scala/io/iohk/praos/domain/package.scala

```diff
@@ -18,9 +18,31 @@ package object domain {
   type StakeDistribution = Map[Key, Stake]
   def StakeDistribution(ps: (Key, Stake)*) = Map[Key, Stake](ps: _*)
 
+  def applyTransaction(transaction: Transaction, stakeDistribution: StakeDistribution): StakeDistribution = {
+    // TODO: Handle logic errors (i.e. that the transaction makes sense in the initial distribution)
+    val resultingSenderStake = stakeDistribution(transaction.senderPublicKey) - transaction.stake
+    val resultingRecipientStake = stakeDistribution(transaction.recipientPublicKey) + transaction.stake
+    val resultingStakeDistribution = stakeDistribution
+      .updated(transaction.senderPublicKey, resultingSenderStake)
+      .updated(transaction.recipientPublicKey, resultingRecipientStake)
+    resultingStakeDistribution
+  }
+
   /**
     * The fraction of stake held by a stakeholder w.r.t. the total stake held by all stakeholders.
     */
   type RelativeStake = Double
+
+  type SlotNumber = Int
+
+  /**
+    * The order is assume from oldest to latest
+    */
+  type Blockchain = List[Block]
```

Comment:
Although this wasn't implemented in this PR, maybe we could add a comment regarding the different meaning that Blockchain has in this context?

Reply:
:+1: